### ISSUE TITLE
Parameters Initialisation in VSTHost

### ISSUE THREAD

**Issue by OzoneGrif** [lordmulder/DynamicAudioNormalizer#5]
Hello, I'm using your super plugin in VSTHost : http://www.hermannseib.com/english/vsthost.htm
It works great, except that it doesn't initialise its parameters on load.
I have to open the parameters box and fiddle with them to set them everytime I open the application.

![capture](https://cloud.githubusercontent.com/assets/10961896/8947805/0bde08c4-35a0-11e5-8a3f-a8863e446d97.PNG)

**Comment by lordmulder** [comment]
Hello,

you need to be aware that there are two "flavors" of VST plug-in's: VST plug-in's that draw their own fancy GUI (graphical user interface), and VST plug-in's that simply expose a number of configuration parameters to the host application. The *Dynamic Audio Normalizer* belongs to the **latter** category.

This means that the configuration dialog that you see in VSTHost is actually created by VSTHost itself. It is **not** created by the Dynamic Audio Normalizer! And it looks totally different in other host applications. For example, in *Adobe Audition*, the Dynamic Audio Normalizer configuration dialog looks like that:

![](http://i.imgur.com/ZAyRlOz.png)

Simply put, the host application can ask the VST plug-in which config parameters are available. Also, for each parameter, the host application can read out the current value or set up a new value. But that's it! All the rest is up to the specific host application! Most host applications will "remember" the config values of a VST plug-in, and it will restore those values when the same VST plug-in is loaded again. Usually the host application will also offer the user a way to save the current config values to a "preset" and to restore a previously saved preset.

**Comment by OzoneGrif** [comment]
Thank you for that detailled answer. I understand now that the problem comes from the host, not the plugin.
